### DISCUSSION TITLE
Visual Feedback to show if the Freeplane window is activated

### DISCUSSION THREAD

**Discussion by euu2021** [freeplane/freeplane#1276] [General]
Something that I do a lot in Freeplane is pasting content from other programs. I spend hours doing this, back and forth:

-1 read content on a PDF or web browser. Copy relevant piece of text.
-2 alternate to FP that is sitting on a secondary monitor or split window
-3 paste content
-4 read a bit more
-5 write a bit on FP

There's something that blocks this flow, though: I never know if the FP window is activated. This is a problem, for example, in the step 5 above, because I don't know if I can start writing right away (I use the FP setting in which a new node is created when I start typing).  I mean, from step 3 to step 4 sometimes I just scroll down the webpage without activating it, so the FP window stays activated, but sometimes I actually click on the webpage.

This situation has been annoying me for years 🤣🤣

I know that Windows has a bit of visual feedback for this, but it is just too minimal to be helpful. Seriously, it's difficult to tell the difference, at a glance, between those two states (I use undocked FP windows, so that's all the visual feedback that I get):
![image](https://github.com/freeplane/freeplane/assets/77707706/9e421f2b-6444-48fb-9af8-9fa3c3a47dad)

So, I think that there are probably a couple ways to solve this. For example:

1.  change the bubble color depending on the state of the window. By bubble color, I mean that bubble around the node that is currently selected.
2. toggle the state of the spotlight depending on the state of the window. So, if the window is inactive, the spotlight turns on; if the window is active, the spotlight turns off. I like this idea because, when the window is inactive, it usually means that the users moved his eyes to another window, so the spotlight is helpful for him to easily locate where he was working, when he moves his eyes back to the FP window.

Any improvement in that direction will be very helpful. 
Thanks,

**Comment by euu2021** [comment]
There is this related problem, that I also face, that maybe could receive a joint solution with the problem above.

When there are multiple map views opened on the same window, there is no way of knowing which one is currently activated. For example, even though I know that this window is activated, I don't know which node will receive the pasted content:

![image](https://github.com/freeplane/freeplane/assets/77707706/5a53f02e-117c-4782-b4ed-e1742833b82a)

**Reply by EdoFro** [reply]
> When there are multiple map views opened on the same window, there is no way of knowing which one is currently activated.

I think this problem deserves a feature request. It may not be urgent, but it is a valid one.

jmho,
edo

**Reply by euu2021** [reply]
Some days ago I created #1277 referring active map view and active window. Now, I split the requests:
-Active window: #1277 
-Active map view:  #1291 

If you have any idea on how this visual feedback could be, please comment.

If I remember correctly, it's difficult to change things in this part of the interface (it's an old library, or something like that): 
![image](https://github.com/freeplane/freeplane/assets/77707706/c49d7ab0-30eb-4238-a0f5-30c3d70958f0)

So, maybe an easier solution is having an option like that: Deactivate the spotlight only on the active map view. 
This would allow, if the user wants, a very clear visual feedback. Exaggerated example, using a colored spotlight:

![image](https://github.com/freeplane/freeplane/assets/77707706/b272bf01-4c5e-4929-8c92-b621fd29591b)

**Reply by EdoFro** [reply]
good idea 👍🏼 
It could also be made by changing the background color of inactive views (or something like that).

**Comment by euu2021** [comment]
Also relevant to have in mind: when using FP in full screen, the OS feedback about the active window is not shown, making the first problem above even more troublesome. So, even in an OS with better visual feedback, the problem is still relevant.

**Comment by euu2021** [comment]
Playing with the settings, I found that some of the "Look and Feel" options in FP hide the accent color of Windows. I changed from Flat Darcula (my favorite) to "Default", and now I can set the top bar to show a color when the window is active:
![explorer_L2dNX1NkbY](https://github.com/freeplane/freeplane/assets/77707706/5b2b27f2-9255-48a0-b0a1-f202d588d237)


This greatly improves the situation. This doesn't help the multiple map views and full screen situation, though (and I can't use Flat Darcula anymore).

**Reply by kenfhill84083** [reply]
@euu2021 Thanks.  I also had to tweak windows 11 "personalization" settings to make title bars show window's accent color show on title bars.
